crates/rustc_codegen_spirv/src/linker/new_structurizer.rs

```diff
@@ -37,6 +44,32 @@ pub fn structurize(
 ) -> Module {
     let mut builder = Builder::new_from_module(module);
 
+    // Get the `OpTypeBool` type (it will only be created if it's missing).
+    let type_bool = builder.type_bool();
+
+    // Find already present `OpConstant{False,True}` (if they're in the module).
+    let mut existing_const_false = None;
+    let mut existing_const_true = None;
+    for inst in &builder.module_ref().types_global_values {
+        let existing = match inst.class.opcode {
+            Op::ConstantFalse => &mut existing_const_false,
+            Op::ConstantTrue => &mut existing_const_true,
+            _ => continue,
+        };
+
+        if existing.is_none() {
+            *existing = Some(inst.result_id.unwrap());
+        }
+
+        if existing_const_false.is_some() && existing_const_true.is_some() {
+            break;
+        }
+    }
+
+    // Create new `OpConstant{False,True}` if they're missing.
+    let const_false = existing_const_false.unwrap_or_else(|| builder.constant_false(type_bool));
+    let const_true = existing_const_true.unwrap_or_else(|| builder.constant_true(type_bool));
```

Comment:
ah, was going to comment on this, but with the new DCE pass happening after the structurizer, this is fine if these aren't used anywhere :P